### PR TITLE
Check for Valid To and From Addresses on eth_call

### LINKED ISSUE #984: Add valid account checks to eth_call
### Problem

Currently `eth_call` doesn't check for the validity of the contract or account from address.

### Solution

check for the validity of the contract or account from address.
If invalid, fail fast.

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by AlfredoG87**
**Description**:
Adding validation for checking if From is a valid account address and To is a valid contract or HTS address

**Related issue(s)**:

Fixes #984 

**Notes for reviewer**:
We are only checking if the "From" address is valid if is different from blank, since there are some Acceptance Tests that imply that "From" could be missing and still expect a response from eth_call
https://github.com/hashgraph/hedera-json-rpc-relay/blob/main/packages/server/tests/acceptance/rpc_batch3.spec.ts#L124

While doing some research and tests, "From" field is indeed [optional](https://docs.alchemy.com/reference/eth-call), however if is included and is invalid, we get the error from consensus node that we are trying to avoid with this PR/Improvement.

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by Nana-EC — DISMISSED** [review]
Looks good.
One test expansion suggestion if it's missing.

File: packages/relay/src/lib/eth.ts [hunk 1]
```diff
@@ -979,6 +979,19 @@ export class EthImpl implements Eth {
...
+    if(call.from) {
+      const fromEntityType = await this.mirrorNodeClient.resolveEntityType(call.from, requestId, [constants.TYPE_ACCOUNT]);
+      if (fromEntityType?.type !== constants.TYPE_ACCOUNT) {
+        throw predefined.NON_EXISTING_ACCOUNT(call.from);
+      }
+    }
+    // Check "To" is a valid Contract or HTS Address
+    const toEntityType = await this.mirrorNodeClient.resolveEntityType(call.to, requestId, [constants.TYPE_TOKEN, constants.TYPE_CONTRACT]);
+    if(!(toEntityType?.type === constants.TYPE_CONTRACT || toEntityType?.type === constants.TYPE_TOKEN)) {
+      throw predefined.NON_EXISTING_CONTRACT(call.to);
```
[the remark this answers is not in the document]
Comment (AlfredoG87) [reply packages/relay/src/lib/eth.ts:992]: Yes, we already had the scenario, but updated the expected respose from old error to new error.
https://github.com/hashgraph/hedera-json-rpc-relay/blob/984-eth_call-check-accounts-validity/packages/server/tests/acceptance/rpc_batch3.spec.ts#L121

File: packages/relay/src/lib/errors/JsonRpcError.ts [hunk 2]
```diff
@@ -171,7 +171,31 @@ export const predefined = {
...
+    });
+  },
+  'NON_EXISTING_ACCOUNT':  (address) => {
+    let message = `Non Existing Account Address: ${address}.`;
+    if (address && address.length) {
+      message = `${message} Expected an Account Address.`;
+    }
+
+    return new JsonRpcError({
+      name: 'Not an Account Address',
```
[the remark this answers is not in the document]
Comment (AlfredoG87) [reply packages/relay/src/lib/errors/JsonRpcError.ts]: Thank you for noticing, little details 😅 
✅  done!

**Review by lukelee-sl — APPROVED** [review]
LGTM

**Review by ebadiere — COMMENTED** [review]
LGTM.  I see a lot of added semicolons added where they were missing.  Thanks for that!

**Review by Nana-EC — APPROVED** [review]

**Pull request merged**